core/src/main/java/org/elasticsearch/cluster/metadata/IndexMetaData.java

```diff
@@ -200,12 +203,19 @@ public static State fromString(String state) {
             .numberOfShards(1).numberOfReplicas(0).build();
 
     public static final String KEY_ACTIVE_ALLOCATIONS = "active_allocations";
+    static final String KEY_VERSION = "version";
+    static final String KEY_SETTINGS = "settings";
+    static final String KEY_STATE = "state";
+    static final String KEY_MAPPINGS = "mappings";
+    static final String KEY_ALIASES = "aliases";
+    static final String KEY_PRIMARY_TERMS = "primary_terms";
 
     private final int numberOfShards;
     private final int numberOfReplicas;
 
     private final Index index;
     private final long version;
+    private final long[] primaryTerms;
```

Comment:
If you use ImmutableOpenIntMap instead (as for activeAllocationIds), you get cluster state diffs on it for free.